.github/workflows/ci_job.yml

```diff
@@ -17,20 +17,21 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "3.X"
+      - name: Install dependencies
```

Comment:
general: minor - to avoid of defining repetitive steps whats ur idea about using YAML anchors is?

```yaml
jobs:
  lint-and-docs:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          submodules: recursive
      - &Set-up-Python
      - name: Set up Python "3.X"
        uses: actions/setup-python@v5
        with:
          python-version: "3.X"
      - &Install-dependencies
      - name: Install dependencies
        run: |
          python3 -m pip install -e ./freva-rest[dev] -e ./freva-data-portal-worker[full]
      - &Generate-self-signed-keys
      - name: Generate self signed keys
        run: |
          python3 run_server.py --gen-certs
      - &Set-up-services
      - name: Set up services
        run: |
          docker-compose -f dev-env/docker-compose.yaml up -d --remove-orphans
      - &Waiting-for-keycloak
      - name: Waiting for keycloak
        run: |
          python run_server.py -w
      - name: Starting the local server
        run: |
          python run_server.py -c api_config.toml --debug --dev -p 7777
      - name: Linting, type checking and docs with tox
        run: tox run-parallel --parallel-no-spinner
  tests:
    runs-on: ubuntu-latest
    strategy:
      max-parallel: 5
      matrix:
        python-version: [3.9, "3.10", "3.11", "3.12"]
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          submodules: recursive
      - <<: *Set-up-Python
      - <<: *Install-dependencies
      - <<: *Generate-self-signed-keys
      - <<: *Set-up-services
      - <<: *Waiting-for-keycloak
      - name: Run tests
        run: tox -e test
      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v3
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          verbose: true

  build-docker-image:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      packages: write
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      - <<: *Generate-self-signed-keys
      - <<: *Set-up-services
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

      - name: Build freva-storage-service image
        uses: docker/build-push-action@v4
        with:
          platforms: linux/amd64
          push: false
          load: true
          tags: freva-rest:latest

      - <<: *Waiting-for-keycloak


      - name: Check databrowser image
        run: python3 dev-env/check-container.py


````